torchinductor/lowering.py

```diff
@@ -111,6 +111,8 @@ def decode_device(device):
         return torch.tensor(0.0).device  # default device
     if isinstance(device, str):
         return torch.device(device)
+    if device.type == "cuda" and device.index is None:
+        return torch.device("cuda", index=torch.cuda.current_device())
```

Comment:
do we guard on current device anywhere ?

Reply:
We don't, though seems fine to do that in another PR.